inc/vcf/fixer.hpp

```diff
@@ -34,125 +34,35 @@ namespace ebi
-        {
-            this->line_number = line_number;
-            this->line = &line; // this will be valid until this function returns
-            error.apply_visitor(*this);
-            this->line = nullptr;
-        }
+        void fix(size_t line_number, std::vector<char> &line, Error &error);
 
-        size_t get_ignored_errors() {return ignored_errors;}
+        size_t get_ignored_errors();
 
       private:
         size_t line_number;
         std::vector<char> *line;
         std::ostream &output;
         size_t ignored_errors;
 
-        virtual void visit(Error &error) override
-        {
-            util::writeline(output, *line);
-            ignored_errors++;
-        }
-        virtual void visit(MetaSectionError &error) override
-        {
-            util::writeline(output, *line);
-            ignored_errors++;
-        }
-        virtual void visit(HeaderSectionError &error) override
-        {
-            util::writeline(output, *line);
-            ignored_errors++;
-        }
-        virtual void visit(BodySectionError &error) override
-        {
-            util::writeline(output, *line);
-            ignored_errors++;
-        }
-        virtual void visit(NoMetaDefinitionError &error)
-        {
-            util::writeline(output, *line);
-            ignored_errors++;
-        }
-        virtual void visit(FileformatError &error) override
-        {
-            util::writeline(output, *line);
-            ignored_errors++;
-        }
-        virtual void visit(ChromosomeBodyError &error) override
-        {
-            util::writeline(output, *line);
-            ignored_errors++;
-        }
-        virtual void visit(PositionBodyError &error) override
-        {
-            util::writeline(output, *line);
-            ignored_errors++;
-        }
-        virtual void visit(IdBodyError &error) override
-        {
-            util::writeline(output, *line);
-            ignored_errors++;
-        }
-        virtual void visit(ReferenceAlleleBodyError &error) override
-        {
-            util::writeline(output, *line);
-            ignored_errors++;
-        }
-        virtual void visit(AlternateAllelesBodyError &error) override
-        {
-            util::writeline(output, *line);
-            ignored_errors++;
-        }
-        virtual void visit(QualityBodyError &error) override
-        {
-            util::writeline(output, *line);
-            ignored_errors++;
-        }
-        virtual void visit(FilterBodyError &error) override
-        {
-            util::writeline(output, *line);
-            ignored_errors++;
-        }
+        virtual void visit(Error &error) override;
+        virtual void visit(MetaSectionError &error) override;
+        virtual void visit(HeaderSectionError &error) override;
+        virtual void visit(BodySectionError &error) override;
+        virtual void visit(NoMetaDefinitionError &error) override;
```

Comment:
Yes, good spot! it's not strictly needed, but it is recommended because it won't allow compilation in the case that it is actually not overriding.

by the way, if we allow further inheriting (we do because we are using `virtual`), we should add a virtual destructor (`virtual ~Fixer() {}`) to avoid memory leaks due to polymorphic use http://stackoverflow.com/questions/461203/when-to-use-virtual-destructors.